storybookjs/storybook
# Enhance the backgrounds addon to support CSS variables.

## Issue
In my case I have mono repo with different frontends. Each frontend should have own theme, but my core component should support all themes. In my stories I want to apply a specific theme. Therefore the change of the background color is not enough. Modern theming should use [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties). 

## What I did
Because of this I extended the background model of the backgrounds addon to support CSS variables. 

The are two ways to provide a theme:
1. Configuration
```js
  backgrounds: {
    default: 'theme-with-css-variables',
    values: [
      {
        name: 'theme-with-css-variables',
        value: '#ff4785',
        variables: {
          '--text-color': '#ff4785',
        },
      },
    ],
  },
```

2. Asset File (CSS)
```js
backgrounds: {
    default: 'theme-with-css-variables-from-asset-file',
    values: [
      {
        name: 'theme-with-css-variables-from-asset-file',
        value: '#ff4785',
        variablesAsset: 'assets/sample-theme-variables.css',
      },
    ],
  },
```
```css
/* sample-theme-variables.css */
:root {
  --text-color: #ff4785;
}
```

Depending on the parameters I create a style element or link element dynamically to apply CSS variables.

I renamed the tooltip and the clear button to make it clearer what is behind the action.


I think this should be a part of the essentials (addon bundle) and I think this should not be a separate addon to avoid confusion with background addon.

I need help if further documentation is necessary. For instance where is the right place to describe this?

## How to test

- Is this testable with Jest or Chromatic screenshots? yes
- Does this need a new example in the kitchen sink apps? yes
- Does this need an update to the documentation? yes

If your answer is yes to any of these, please make sure to include it in your PR.

---

I extended the existing story file for background addon stories:
`yarn storybook` in the `storybook/examples/official-storybook` folder
http://localhost:9011/?path=/story/addons-backgrounds--with-css-variables
http://localhost:9011/?path=/story/addons-backgrounds--with-css-variables-from-asset




### Screenshot
![image](https://user-images.githubusercontent.com/1221055/95013762-aa516000-0642-11eb-8408-c27f8a49ea13.png)




Please assign [**hacktoberfest-accepted**](https://hacktoberfest.digitalocean.com/) label to this PR.
